### PR TITLE
Add workaround for Swift + lld bug

### PR DESCRIPTION
**Pull request by keith**
This is what SwiftPM does now as well.

### CODE REVIEW

File: swift/internal/swift_autoconfiguration.bzl [hunk 1]
```diff
@@ -130,6 +131,27 @@ def _check_use_response_files(repository_ctx, swiftc_path, temp_dir):
         "@{}".format(param_file),
     )
 
+def _check_supports_lld_gc_workaround(repository_ctx, swiftc_path, temp_dir):
+    """Returns True if `swiftc` supports implementation-only imports."""
```
Comment (brentleyjones) [inline swift/internal/swift_autoconfiguration.bzl:135]: Wrong comment?

Comment (keith) [reply swift/internal/swift_autoconfiguration.bzl:135]: fixed

File: swift/internal/swift_autoconfiguration.bzl [hunk 2]
```diff
@@ -130,6 +131,27 @@ def _check_use_response_files(repository_ctx, swiftc_path, temp_dir):
         "@{}".format(param_file),
     )
 
+def _check_supports_lld_gc_workaround(repository_ctx, swiftc_path, temp_dir):
```
Comment (brentleyjones) [inline swift/internal/swift_autoconfiguration.bzl:134]: How do we enable this only if someone is using lld (to match what the Swift driver is doing)?

Comment (keith) [reply swift/internal/swift_autoconfiguration.bzl:134]: if lld is found the cc toolchain always uses it

Comment (keith) [reply swift/internal/swift_autoconfiguration.bzl:134]: https://github.com/bazelbuild/bazel/blob/ccfeeff06cdc0266b92781213bbdf33c4288b1c5/tools/cpp/unix_cc_configure.bzl#L440

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**